VK/VK/ShowGriendPhotoScreen/ShowFriendPhotoScreenViewController.swift

```diff
@@ -0,0 +1,169 @@
+            slide.photoImageView.isUserInteractionEnabled = true
+            slides.append(slide)
+        }
+
+        return slides
+    }
+
+    private func configureScrollView() {
+        guard let slidesForScroll = slides else { return }
+        showPhotoScrollView.frame = CGRect(x: 0, y: 0, width: view.frame.width, height: view.frame.height)
+        showPhotoScrollView.contentSize = CGSize(
+            width: view.frame.width * CGFloat(slidesForScroll.count),
+            height: view.frame.height
+        )
+        showPhotoScrollView.isPagingEnabled = true
+
+        for index in 0 ..< slidesForScroll.count {
+            slidesForScroll[index]
+                .frame = CGRect(
+                    x: view.frame.width * CGFloat(index),
+                    y: 0,
+                    width: view.frame.width,
+                    height: view.frame.height
+                )
+            showPhotoScrollView.addSubview(slidesForScroll[index])
+        }
+    }
+
+    private func configurePageControlView() {
+        pageControllView.numberOfPages = slides?.count ?? 0
+        pageControllView.currentPage = 0
+        view.bringSubviewToFront(pageControllView)
+    }
+
+    private func configureTapHandler() {
+        tapHandler = {
+            self.dismiss(animated: true)
+        }
+    }
+
+    @objc private func gestoreAction() {
+        guard let action = tapHandler else { return }
+        action()
+    }
+}
+
+// MARK: - UIScrollViewDelegate
+
+extension ShowFriendPhotoScreenViewController: UIScrollViewDelegate {
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        guard let slides = slides else { return }
+        let pageIndex = round(scrollView.contentOffset.x / view.frame.width)
+        pageControllView.currentPage = Int(pageIndex)
+
+        let maximumHorizontalOffset: CGFloat = scrollView.contentSize.width - scrollView.frame.width
+        let currentHorizontalOffset: CGFloat = scrollView.contentOffset.x
+
+        let maximumVerticalOffset: CGFloat = scrollView.contentSize.height - scrollView.frame.height
+        let currentVerticalOffset: CGFloat = scrollView.contentOffset.y
+
+        let percentageHorizontalOffset: CGFloat = currentHorizontalOffset / maximumHorizontalOffset
+        let percentageVerticalOffset: CGFloat = currentVerticalOffset / maximumVerticalOffset
+
+        let percentOffset = CGPoint(x: percentageHorizontalOffset, y: percentageVerticalOffset)
+
+        if percentOffset.x > 0, percentOffset.x <= 0.25 {
+            slides[0].photoImageView
+                .transform = CGAffineTransform(
+                    scaleX: (0.25 - percentOffset.x) / 0.25,
+                    y: (0.25 - percentOffset.x) / 0.25
+                )
+            slides[1].photoImageView
+                .transform = CGAffineTransform(scaleX: percentOffset.x / 0.25, y: percentOffset.x / 0.25)
+
+        } else if percentOffset.x > 0.25, percentOffset.x <= 0.50 {
+            slides[1].photoImageView
```

Comment:
```suggestion
            slides[1].photoImageView
```
небезопасное обращение